languageFactories/cwl-v1-0/src/main/scala/languages/cwl/CwlV1_0LanguageFactory.scala

```diff
@@ -32,18 +31,21 @@ class CwlV1_0LanguageFactory(override val config: Config) extends LanguageFactor
                                  ioFunctions: IoFunctionSet,
                                  importResolvers: List[ImportResolver]): IOChecked[ValidatedWomNamespace] = {
 
-    def parse(): IOChecked[Cwl] = source.workflowUrl match {
-      case Some(url) => for {
-        reference <- fromEither[IO](CwlReference.fromString(url).map(Right(_)).getOrElse(Left(NonEmptyList.one(s"Invalid workflow reference: $url")))): IOChecked[CwlReference]
-        parsed <- CwlDecoder.decodeCwlReference(reference.changePointer(source.workflowRoot))
-      } yield parsed
-      case None =>
-        CwlDecoder.decodeCwlString(
-          workflowSource,
-          source.importsZipFileOption.map(File.newTemporaryFile().appendByteArray(_)),
-          source.workflowRoot,
-          "cwl_temp_file_" + workflowIdForLogging.toString
-        )
+    def parse(): IOChecked[Cwl] = {
+      val decodedCwlReference =
+        for {
+          url <- source.workflowUrl
+          reference <- CwlReference.fromString(url)
+        } yield CwlDecoder.decodeCwlReference(reference.changePointer(source.workflowRoot))
+
+      decodedCwlReference
+        .getOrElse(
+          CwlDecoder.decodeCwlString(
+            workflowSource,
+            source.importsZipFileOption.map(File.newTemporaryFile().appendByteArray(_)),
+            source.workflowRoot,
+            "cwl_temp_file_" + workflowIdForLogging.toString
```

Comment:
Maybe we should extract this string in variable?

Reply:
It is not exactly what you have wanted, but now this string is a named variable and also I've replaced string concatenation to a string interpolation.
I believe now it is much better and there is no need to extract it to a separate variable.